services/src/main/java/org/exoplatform/news/service/impl/NewsTargetingServiceImpl.java

```diff
@@ -100,16 +118,28 @@ public void saveNewsTarget(String newsId, List<String> targets, String currentUs
   }
 
   @Override
-  public List<MetadataItem> getNewsTargetItemsByTargetName(String targetName, long offset, long limit) {
+  public List<MetadataItem> getNewsTargetItemsByTargetName(String targetName, org.exoplatform.services.security.Identity currentIdentity, long offset, long limit) {
+    if (!canPublishNews(currentIdentity)) {
+      throw new IllegalArgumentException("User " + currentIdentity.getUserId() + " not authorized to publish news");
+    }
     return metadataService.getMetadataItemsByMetadataNameAndTypeAndObject(targetName, METADATA_TYPE.getName(), NewsUtils.NEWS_METADATA_OBJECT_TYPE, offset, limit);
   }
 
   @Override
-  public void deleteNewsTargets(String newsId) {
+  public void deleteNewsTargets(String newsId, String currentUserId) {
+    org.exoplatform.services.security.Identity currentIdentity = NewsUtils.getUserIdentity(currentUserId);
     NewsTargetObject newsTargetObject = new NewsTargetObject(NewsUtils.NEWS_METADATA_OBJECT_TYPE, newsId, null);
+    if (!canPublishNews(currentIdentity)) {
+      throw new IllegalArgumentException("User " + currentIdentity.getUserId() + " not authorized to publish news");
+    }
     metadataService.deleteMetadataItemsByMetadataTypeAndObject(METADATA_TYPE.getName(), newsTargetObject);
   }
 
+  @Override
+  public boolean canPublishNews(org.exoplatform.services.security.Identity currentIdentity) {
```

Comment:
define this method in NewsUtils as static method to be used from different locations, to be removed also from NewsServiceImpl.java